batch.go

```diff
@@ -0,0 +1,129 @@
+package main
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func splitpath(path string) (string, string) {
+	path_components := strings.Split(path, "/")
+	basename := path_components[len(path_components)-1]
+	dirname := strings.Join(path_components[:len(path_components)-1], "/")
+	return basename, dirname
+}
+
+func is_hidden_file(path string) bool {
+	basename, _ := splitpath(path)
+	return strings.HasPrefix(basename, ".")
+}
+
+func aggregate_to_file(decks []*Deck, deckname string, output_path string) {
+	var archetypical_deck *Deck
+	if len(decks) == 1 {
+		// trivially, a single deck's aggregate is itself
+		archetypical_deck = decks[0]
+	} else {
+		var err error
+		archetypical_deck, err = aggregate(decks)
+		if err != nil {
+			log.Fatalln("Couldn't create archetype for", deckname+":", err)
+		}
+	}
+
+	err := os.MkdirAll(output_path, 0755)
+	if err != nil {
+		log.Fatalln("Couldn't create output directory", output_path+":", err)
+	}
+	filename := output_path + "/" + deckname + "-aggregate.txt"
+	err = ioutil.WriteFile(filename, []byte(archetypical_deck.String()), 0644)
+	if err != nil {
+		log.Fatalln("Couldn't write output file", output_path+":", err)
+	}
+}
+
+func strip_trailing_slash(s *string) {
+	l := len(*s)
+	if (*s)[l-1] == '/' {
+		*s = (*s)[:l-2]
+	}
+}
+
+func batch(top_path string, output_path string, verbose bool) {
+	decks_processed := 0
+	aggregate_decks_created := 0
+
+	strip_trailing_slash(&top_path)
+	strip_trailing_slash(&output_path)
+
+	var deck_dir, deckname string
+	decks := make([]*Deck, 0)
+
+	filepath.Walk(top_path, func(path string, info os.FileInfo, _ error) error {
```

Comment:
I'm not crazy about the ignored error and log.Fatals, but if this code is serving your needs they can stay as-is.

Just be warned I'll likely change those to return errors up to the entry point if/when I touch this again.